### ISSUE TITLE
travis-ci: descrease pressure on Travis-CI machines

### ISSUE THREAD

**Issue by Totktonada** [tarantool/tarantool#4410]
The old plan was [here](https://github.com/tarantool/tarantool/issues/4156#issuecomment-483435050). I short it and modify a bit:

In short:

* Disable LTO / ASAN / static build in Travis-CI.
* Disable short-term branches testing on Travis-CI (don't quite sure however). Or maybe at least look whether to move coverage collecting / reporting to GitLab-CI: I mean, leave debug build, but w/o coverage to speed the job up.
* Consider moving deploying to GitLab CI (don't sure however). Or maybe disable pre-upload testing in Travis-CI and leave it only for GitLab CI.

**Comment by Totktonada** [comment]
Removed from Travis-CI:

- OS X 10.13 (High Sierra); it fails in brew update:

```
==> Summary

🍺  /usr/local/Cellar/zlib/1.2.11: 12 files, 373.5KB

Error: automake 1.16.1 is already installed

To upgrade to 1.16.1_1, run brew upgrade automake.
```

https://travis-ci.org/tarantool/tarantool/jobs/574881975

Don't sure about a precise reason.

- LTO builds.
- ASAN build.

**Comment by Totktonada** [comment]
Most of Travis CI jobs was moved to GitLab CI, but the last ones were moved to GitHub Actions (in the scope of #5294). The problem was finally closed in 7aa0b018726a651601c758228c8ad965135a29d0 ([full history](https://github.com/tarantool/tarantool/commits/master/.travis.yml)).

We're going to use GitHub Actions as the signle CI provider (so, moving from GitLab CI to GitHub Actions now).
